cogeo_mosaic/overviews.py

```diff
@@ -124,80 +133,200 @@ def create_low_level_cogs(
+                            rasterio.open(tmpfile.name, "w", **params)
                         )
+
+                    def _get_tile(wind):
+                        idx, window = wind
+                        x = extrema["x"]["min"] + idx[1]
+                        y = extrema["y"]["min"] + idx[0]
+                        t = mercantile.Tile(x, y, base_zoom)
+
+                        kds = set(find_quadkeys(t, mosaic_quadkey_zoom))
+                        if not mosaic_quadkeys.intersection(kds):
+                            return window, None, None
+
+                        assets = mosaic.tile(*t)
+                        if not assets:
+                            return window, None, None
+
+                        tile, mask = mosaic_tiler(
+                            assets,
+                            x,
+                            y,
+                            base_zoom,
+                            _tile,
+                            tilesize=tilesize,
+                            pixel_selection=pixel_method(),
+                        )
+
+                        return window, tile, mask
+
+                    with futures.ThreadPoolExecutor(max_workers=threads) as executor:
+                        future_work = [
+                            executor.submit(_get_tile, item) for item in blocks
+                        ]
+                        with click.progressbar(
+                            futures.as_completed(future_work),
+                            length=len(future_work),
+                            show_percent=True,
+                        ) as future:
+                            for res in future:
+                                pass
+
+                    for f in _filter_futures(future_work):
+                        window, tile, mask = f
+                        if tile is None:
+                            continue
+
+                        tmp_dst.write(tile, window=window)
+                        if info["nodata_type"] == "Mask":
+                            tmp_dst.write_mask(mask.astype("uint8"), window=window)
+
+                    min_tile_size = tilesize = min(
+                        int(output_profile["blockxsize"]),
+                        int(output_profile["blockysize"]),
+                    )
+                    overview_level = get_maximum_overview_level(
+                        tmp_dst.width, tmp_dst.height, minsize=min_tile_size
+                    )
+                    overviews = [2 ** j for j in range(1, overview_level + 1)]
+                    tmp_dst.build_overviews(overviews)
+                    copy(tmp_dst, output_path, copy_src_overviews=True, **params)
+
+
+def create_high_level_cogs(
+    mosaic_path: str,
+    output_path: str,
+    output_profile: Dict,
+    method: str = "first",
+    config: Dict = None,
+    threads=1,
+    in_memory: bool = True,
+) -> None:
+    """
+    Create WebOptimized Overview COG from a mosaic definition file.
```

Comment:
I'd suggest a better docstring for what the difference between `high_level` and `low_level` is